### PR TITLE
Wip memory

### PR DESCRIPTION
**Pull request by DoDucAnh2810**

### CODE REVIEW

File: memory.c [hunk 1]
```diff
@@ -68,30 +90,136 @@ void memory_destroy(memory mem) {
...
+        return INDEX_OUT_OF_RANGE; 
+    } 
+
+    uint16_t aux = 0;
+    be = (be > 0) ? MEM_BIG_ENDIAN : MEM_LITTLE_ENDIAN; // cast be to bool
+    switch ((uint8_t)(be == is_big_endian()))   // check if be is sytem convention
+    {
+    case 0: // false inverse value
+        aux |= ((uint16_t)(mem->data[address++])) << 8;
+        aux |= (uint16_t)(mem->data[address]);
```
Comment (DoDucAnh2810) [inline memory.c:119]: Wrong inverse of endian

Comment (DoDucAnh2810) [reply memory.c:119]: You didn't change anything with your inverse.
You must push what we see first to the right not the left.
byte1-byte2 --> byte2-byte1 (we see byte1 first at address then byte2 at address + 1)
So it is like:
`aux |= (uint16_t)(mem->data[address]);
aux |= (unint16_t)(mem->data[address+1]) << 8;`

File: memory.c [hunk 2]
```diff
@@ -68,30 +90,136 @@ void memory_destroy(memory mem) {
...
+        return INDEX_OUT_OF_RANGE; 
+    } 
+
+    uint32_t aux = 0;
+    be = (be > 0) ? MEM_BIG_ENDIAN : MEM_LITTLE_ENDIAN; // cast be to bool
+    switch ((uint8_t)(be == is_big_endian()))   // check if be is system convention
+    {
+    case 0: // false inverse value
+        for (int i = 0; i < 4; i++) {
+            aux |= ((uint32_t)(mem->data[address+i])) << ((3-i)*8);
```
Comment (DoDucAnh2810) [inline memory.c:146]: Wrong inverse of endian

File: memory.c [hunk 3]
```diff
@@ -68,30 +90,136 @@ void memory_destroy(memory mem) {
...
+        error("word address out of range\n"); 
+        return INDEX_OUT_OF_RANGE; 
+    }
+
+    be = (be > 0) ? MEM_BIG_ENDIAN : MEM_LITTLE_ENDIAN; // cast be to bool
+    switch ((uint8_t)(be == is_big_endian()))   // check if be is system convention
+    {
+    case 0: // false inverse value
+        for (int i = 0; i < 4; i++) {
+            mem->data[address+i] = (uint8_t)(value >> (3-i)*8);
```
Comment (DoDucAnh2810) [inline memory.c:212]: Wrong inverse of endian

**Review by TheAmazingSui — APPROVED** [review]
I'm too tired to check but it seems to work

**Review by CodeNachos — APPROVED** [review]
it just work.
source: my guts, believe me

**Pull request merged**